neuralforecast/common/_base_recurrent.py

```diff
@@ -49,13 +49,10 @@ def __init__(self,
         self.learning_rate = learning_rate
 
         # Scaler
-        # TODO: clean hack and add correct use of self.scaler
         if scaler_type is None:
-            self.normalize = False
             self.scaler = None
         else:
-            self.normalize = True
-            self.scaler = TemporalNorm(scaler_type=scaler_type)
+            self.scaler = TemporalNorm(scaler_type=scaler_type, dim=-1) # Time dimension is -1.
```

Comment:
It is unclear to me that we want to set the default of `BaseRecurrent` to normalize the autoregressive channel.

We are currently using `input_size` in the `TimeSeriesDataset` to generate autoregresive channels in the last dimension and normalizing them, but a more classical approach would set `input_size=1` and apply the normalizer to the windows in dimension 1:
\# [N,C,T,L=1+H] -> [N,T,C,L=1+H] -> [N,T,C,H]

We need to rethink this.